kong/runloop/balancer.lua

```diff
@@ -752,6 +752,17 @@ end
 -- which do not support sending it)
 -- @return true on success, nil+error message+status code otherwise
 local function execute(target, ctx)
+  do -- Check for KONG_ORIGINS override
+    local origin_key =
+      target.scheme .. "://" .. utils.format_host(utils.normalize_ip(target))
```

Comment:
Something caught my eye, not sure what I may have missed: `utils.normalize_ip()` expects a string, but `target` is a table here, correct?